docs/package-deletion-policies.md

```diff
@@ -0,0 +1,210 @@
+They can delete the original repository, but a copy of the project will remain on the proxy server.
+Instead, Go uses a [retract directive](https://go.dev/ref/mod#go-mod-file-retract), which is an example of a [package deletion alternative](#deletion-alternatives).
+
+### Test Instances and Staging Environments
+
+Another architecture decision related to a package deletion policy is the use of a separate test instance or a staging environment.
+
+For example PyPI maintains [TestPyPI](https://test.pypi.org/) to allow users to try distribution tools and processes without affecting the real PyPI.
+A test instance does not necessarily eliminate the need for a package deletion policy because mistakes can still be made, but it does allow for users to experiment without leaving junk packages on the main instance.
+
+## Deletion Alternatives
+
+There are a number of reasons that a person would want to delete a package.
+Package deletion is a good solution for some of those reasons (publishing the wrong package, making a typo, experimenting with the package manager), but not necessarily for others.
+
+What if a package (or version series) is no longer in active development?
+What if a specific version has a security issue?
+In these cases, the community should be informed of the issues, and possibly prevented from using a specific version, but deleting the package outright is not a tenable solution.
+So what can package managers do in these cases?
+
+### Yanking
+
+"Yanking" refers to soft deleting a compromised package version.
+The version still exists, but by marking the version as yanked, installers will ignore the version unless it is the only release that matches a version specifier.
+If the yanked version is explicitly required, it will be installed and will not break the build.
+Maintainers should consider yanking a release if it is broken, unstable, or contains a very severe vulnerability.
+Yanking will not completely guarantee that no one is using the yanked version, but it will reduce the installer preference to install it.
+If an ecosystem allows for version yanking, it is best practice to also allow maintainers to provide a reason.
+Users who are explicitly requiring the yanked version can then appropriately prioritize moving to a non-yanked version.
+
+Some version yanking policies use slightly different terminology but are functionally very similar.
+For example, Go uses the term "retract" and the retract directive can indicate that a package version (or version range) should not be used.
+
+Regardless of the term used, the following sets these policies apart from a deletion policy in the following ways:
+
+1. The yanked version is still available.
+2. The installer will try to avoid using yanked versions.
+3. The package consumer can make an informed decision about whether to use the yanked version.
+4. Yanking a version will not break a build.
+
+### Deprecation
```

Comment:
You could maybe lump end of life status as part of this. In many contexts the two are used interchangeably but on ecosystem to ecosystem. For the purposes of this document, I think they can be lumped together to improve comprehension.